smartsim/_core/control/controller.py

```diff
@@ -584,7 +598,11 @@ def _launch_step(
         # if the completed job does exist and the entity passed in is the same
         # that has ran and completed, relaunch the entity.
         elif completed_job is not None and completed_job.entity is entity:
-            self.symlink_output_files(job_step, entity)
+            # make sure it's not an _AnonymousBatchJob
+            if not isinstance(entity, Model) or (
```

Comment:
we're repeating the same check twice. can we do it once before the if/else starts and just use the once result?

Reply:
moved calls to symlink out of `launch_step` so this is no longer necessary (in this particular case, that is).